src/ripple/protocol/impl/Feature.cpp

```diff
@@ -18,57 +18,73 @@
 //==============================================================================
 
 #include <BeastConfig.h>
+#include <ripple/protocol/Feature.h>
 #include <ripple/basics/contract.h>
 #include <ripple/protocol/digest.h>
-#include <ripple/protocol/Feature.h>
 
 #include <boost/container/flat_map.hpp>
 
 #include <cstring>
 
 namespace ripple {
 
-static
-uint256
-feature (char const* s, std::size_t n)
-{
-    sha512_half_hasher h;
-    h(s, n);
-    return static_cast<uint256>(h);
-}
-
-static
-uint256
-feature (std::string const& name)
-{
-    return feature(name.c_str(), name.size());
-}
-
-static std::vector<uint256> const features = [] {
-    std::vector<uint256> result;
-    result.reserve(detail::numFeatures);
-    for (std::string const& s : detail::featureNames)
-        result.push_back(feature(s));
-    return result;
-}();
+static constexpr char const* const featureNames[] =
+    {"MultiSign",
+     "Tickets",
+     "TrustSetAuth",
+     "FeeEscalation",
+     "OwnerPaysFee",
+     "CompareFlowV1V2",
+     "SHAMapV2",
+     "PayChan",
+     "Flow",
+     "CompareTakerFlowCross",
+     "FlowCross",
+     "CryptoConditions",
+     "TickSize",
+     "fix1368",
+     "Escrow",
+     "CryptoConditionsSuite",
+     "fix1373",
+     "EnforceInvariants"};
+
+static_assert (detail::numFeatures() ==
+    sizeof (featureNames) / sizeof (featureNames[0]), "Update numFeatures");
+
+static std::vector<uint256> const features =
+    [] {
+        std::vector<uint256> result;
+        result.reserve(detail::numFeatures());
+
+        auto feature = [] (char const* name)
+        {
+            sha512_half_hasher h;
+            h (name, std::strlen (name));
+            return static_cast<uint256>(h);
+        };
+
+        for (char const* const name : featureNames)
+            result.push_back(feature (name));
+        return result;
+    }();
```

Comment:
The three initializer for `features`, `featureToIndex`, and `nameToFeature` are closely related. Let's consider packaging all three collections into a struct and using a constructor to clean this up. Commit: https://github.com/seelabs/rippled/commit/a3b0cba139e5b397cd8bef87257ceca8d340cca9

Note: That commit also changes `featureToBitsetIndex` to use a ref instead of a value